build/build-parse-trees.js

```diff
@@ -58,6 +58,12 @@ names.forEach(function(name) {
   });
   print('   */');
   print('  constructor(%s) {', paramNames.join(', '));
+  if (paramNames[0] === 'location') {
+    print('if (location) {');
```

Comment:
What's the perf implication?

Maybe move this into a function so it is not repeated all over.

This should be `location !== null` since passing `undefined` is an error.